api/src/test/java/org/openmrs/obs/AbstractHandlerTest.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.obs;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+
+import org.apache.commons.io.FilenameUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.openmrs.Obs;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.obs.handler.AbstractHandler;
+import org.openmrs.util.OpenmrsUtil;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ AbstractHandler.class, OpenmrsUtil.class, Context.class })
+
+public class AbstractHandlerTest {
+	
+	private final String FILENAME = "mytxtfile.txt";
+	
+	@Mock
+	private AdministrationService administrationService;
+	
+	@Rule
+	public TemporaryFolder complexObsTestFolder = new TemporaryFolder();
+	
+	@Test
+	public void getOutputFileToWrite_shouldNeverOverwritePreviousFiles() throws IOException {
+		String content1 = "A";
+		String content2 = "B";
+		AbstractHandler handler = new AbstractHandler();
+		
+		File previousFile = null;
+		File currentFile = null;
+		
+		// Mocked methods
+		mockStatic(Context.class);
+		when(Context.getAdministrationService()).thenReturn(administrationService);
+		when(administrationService.getGlobalProperty(any())).thenReturn(complexObsTestFolder.newFolder().getAbsolutePath());
+		
+		// Construct 102 Obs and save them, alternating between 'A' and 'B'
+		for (int i = 0; i <= 101; i++) {
+			String currentData;
+			String previousData;
+			
+			if (i % 2 == 0) {
+				currentData = content1;
+				previousData = content2;
+			} else {
+				currentData = content2;
+				previousData = content1;
+			}
+			
+			ComplexData complexData = new ComplexData(FILENAME, currentData);
+			
+			Obs obs = new Obs();
+			obs.setComplexData(complexData);
+			
+			currentFile = handler.getOutputFileToWrite(obs);
+			
+			// Write the file
+			try (BufferedWriter fout = new BufferedWriter(
+			        new OutputStreamWriter(new FileOutputStream(currentFile), StandardCharsets.UTF_8))) {
+				fout.write(currentData);
+			}
+			
+			// Read the file to check if it wrote properly
+			try (BufferedReader fin = new BufferedReader(
```

Comment:
can we make use of apache commons-io `FileUtils.contentEquals` ?

Reply:
@teleivo For one case, I was able to use contentEquals without having to rewrite most of the test. I think it should be fine as it is now.